Microsoft.DotNet.Interactive/KernelBase.cs

```diff
@@ -251,7 +256,21 @@ public void RegisterForDisposal(IDisposable disposable)
             IKernelCommand command,
             KernelInvocationContext context) => context.HandlingKernel = this;
 
-        public void Dispose() => _disposables.Dispose();
+        protected virtual void Dispose(bool disposing)
+        {
+            if (_disposables.IsDisposed)
```

Comment:
Should `_disposables` be responsible for checking `IsDisposed` in its `Dispose()` method, and no-oping? That way `KernelBase` can just call `Dispose()` on the things it needs to dispose.

Reply:
It probably does, I'm doing this as a sanity check.